### PR TITLE
ci: tekton configurations for PR build pipeline

### PR DESCRIPTION
**Pull request by nithinputhenveettil**
This PR adds Tekton configurations for the PR build pipeline. Eventually, we will replace the existing CircleCI configuration with this.

### CODE REVIEW

File: .tekton/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Tekton Setup on a Cluster
+
+- You need access to a cluster with full admin privileges.
+- Allocate enough RAM and CPU so that all the pods, including sidecar pods, will run smoothly on a single node.
+- Add multiple nodes to increase parallel pipeline runs.
+
+### Tekton Setup
+
+```sh
+# Install pipeline
```
Comment (sanojsubran) [inline .tekton/README.md:12]:
```suggestion
# Install Tekton pipelines
```

File: .tekton/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+
+# Install GitHub trigger
```
Comment (sanojsubran) [inline .tekton/README.md:15]:
```suggestion
# Install Tekton triggers
```

File: .tekton/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/triggers/latest/release.yaml
+
+# Install GitHub interceptor
```
Comment (sanojsubran) [inline .tekton/README.md:18]:
```suggestion
# Install Tekton interceptors
```

File: .tekton/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/triggers/latest/interceptors.yaml
+
+# Install dashboard - full installation is needed to make changes in the pipeline, such as re-running a pipeline run or deleting a pipeline run.
```
Comment (sanojsubran) [inline .tekton/README.md:21]:
```suggestion
# Install Tekton dashboard - full installation is needed for read/write capabilities. eg: to make changes in the pipeline, such as re-running a pipeline run or deleting a pipeline run. 
```

File: .tekton/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+# Install GitHub trigger
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/triggers/latest/release.yaml
+
+# Install GitHub interceptor
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/triggers/latest/interceptors.yaml
+
+# Install dashboard - full installation is needed to make changes in the pipeline, such as re-running a pipeline run or deleting a pipeline run. 
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/dashboard/latest/release-full.yaml
+
+# Monitor installation
```
Comment (sanojsubran) [inline .tekton/README.md:24]:
```suggestion
```

File: .tekton/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+# Make sure all pods are in the ready state before proceeding further.
```
Comment (sanojsubran) [inline .tekton/README.md:25]:
```suggestion
# Make sure all pods are in the ready state before proceeding further by issuing the following command.
```

File: .tekton/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+kubectl get pods --namespace tekton-pipelines --watch
+
+# Access the dashboard
+kubectl proxy
+```
+
+- If you have successfully completed the above steps, you can access the Tekton Dashboard from [here](http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/)
```
Comment (sanojsubran) [inline .tekton/README.md:32]:
```suggestion
- If you have successfully completed the above mentioned steps, you should be able to access the Tekton Dashboard from [here](http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/)
```

File: .tekton/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+
+## Install Tekton pipelines for Go Tracer
+- You will find all the required YAML configurations in the `.tekton` folder on the go-tracer repo. This includes all the required tasks, pipelines, and GitHub triggers, etc.
```
Comment (sanojsubran) [inline .tekton/README.md:36]:
```suggestion
- You will find all the required YAML configurations in the `.tekton` folder, present in the root directory of the go-tracer repo. This includes all the required tasks, pipelines, and GitHub triggers, etc.
```

File: .tekton/README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+### Prerequisites before applying the YAML files
+
+- You need to create three secrets to run the Go tracer pipelines:
+    1. **GitHub bot token** - You need a GitHub bot token with write access to the repo. This is for sending commit statuses.
+    2. **GitHub Webhook Secret** - Create a very long random secret. You need to add this to the GitHub UI when creating a webhook for PR events.
+    3. **Cosmos URL and Secret** - This is for running azcosmos integration tests.
+
+- Once you have access to the above secrets, please replace them in the `secrets.yaml` file.
```
Comment (sanojsubran) [inline .tekton/README.md:45]:
```suggestion
- Once you have access to the above secrets, replace them in the `secrets.yaml` file.
```

File: .tekton/README.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+- You need an ingress controller for the GitHub Webhook to come through.
+- Please replace the `ingressClassName` and ingress domain or subdomain URL in the `github-webhook-ingress.yaml` file.
```
Comment (sanojsubran) [inline .tekton/README.md:47]:
```suggestion
- Replace the `ingressClassName` and ingress domain or subdomain URL in the `github-webhook-ingress.yaml` file.
```

File: .tekton/README.md (hunk 5 above)
Comment (nithinputhenveettil) [reply .tekton/README.md:24]: Can we keep this line ?

Comment (nithinputhenveettil) [reply .tekton/README.md:24]: Aah got it, will delete.

File: .tekton/README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+- Make sure you create a GitHub webhook for the `Pull requests` events in the settings tab of the repo. Please add the previously created webhook secret and `<<ingress_url/hooks>>` as the Payload URL in the appropriate place when creating the webhook.
+
+### Installation
+- Once you are ready with the above steps, please use the below command to apply the YAML files.
+```sh
+sh deploy.sh
+```
+- Congrats! You have successfully configured everything. You will see a status has been posted for the Tekton runs whenever a new PR gets created.
```
Comment (sanojsubran) [inline .tekton/README.md:54]:
```suggestion
- Congrats! You have successfully configured Tekton CI pipeline for Go Tracer. You will see a status posted in Github for the Tekton runs, whenever a new PR is created.
```

File: .tekton/README.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+## How to debug/re-run a pipeline run
+
+- You will find the Tekton dashboard URL for a specific pipeline run from the `details` section of the commit status.
+-You can access the dashboard once you have set up and authenticated the cluster in your local and using the `kubectl proxy` command.  How to do this using IBM Cloud, you can see the documentation [here](https://cloud.ibm.com/docs/containers?topic=containers-access_cluster#access_public_se) for help.
```
Comment (sanojsubran) [inline .tekton/README.md:59]:
```suggestion
- You can access the Tekton dashboard if you had set up the `ibmcloud` cli and authenticated the cluster in your local machine, by using the `kubectl proxy` command. For detailed information on accessing the IBM Cloud cluster via `ibmcloud` cli, you can refer to this [documentation](https://cloud.ibm.com/docs/containers?topic=containers-access_cluster#access_public_se).
```

File: .tekton/README.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+- You can see the logs or re-run the pipeline run once you have access to the dashboard.
```
Comment (sanojsubran) [inline .tekton/README.md:60]:
```suggestion
- Once you have access to the dashboard, you can see the logs for each run and will be able to re-run the `PipelineRun` .
```

File: .tekton/README.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+- You will see that the statuses in the PR will update once you initiate a re-run.
```
Comment (sanojsubran) [inline .tekton/README.md:61]:
```suggestion
- The status of the Tekton CI pipeline run for the PR will be updated once you initiate a re-run.
```

**Review by sanojsubran — APPROVED** [review]
LGTM

**Comment by Angith** [comment]
LGTM. Great effort and Good Work ⭐

**Review by Angith — APPROVED** [review]

**Pull request merged**